Doc/library/typing.rst

```diff
@@ -661,6 +664,7 @@ These can be used as types in annotations using ``[]``, each having a unique syn
    Optional type.
 
    ``Optional[X]`` is equivalent to ``X | None`` (or ``Union[X, None]``).
+   ``X | None`` is prefered in readability in most cases.
```

Comment:
I think `Optional` was still being debated.

Reply:
My opinion is that `X | None` is better than `Optional`:
1. It does not create a separate concept. It is just a union with `None`
2. It does not "look" like "optional argument" to some users

But, this is not a strong opinion. So, I can do both 🙂

Reply:
I also prefer `X | None` over `Optional`

Reply:
`Optional` is also more problematic than `Union`, because it's a constant cause for confusion.

Reply:
You could frame this to avoid adding too much opinion:
```
``X | None`` is equivalent to ``Optional[X]`` and ``Union[X, None]``.
```